nixos/modules/programs/singularity.nix

```diff
@@ -61,7 +61,12 @@ in
     };
     enableSuid = mkOption {
       type = types.bool;
-      default = true;
+      # SingularityCE requires SETUID for most things. Apptainer prefers user
+      # namespaces, e.g. `apptainer exec --nv` would fail if built
+      # `--with-suid`:
+      # > `FATAL: nvidia-container-cli not allowed in setuid mode`
+      default = cfg.packageOverriden.pname != "apptainer";
+      defaultText = literalExpression ''config.services.singularity.packageOverriden.pname != "apptainer"'';
```

Comment:
Update: We should get it from `cfg.package`, since we are deciding on how to build `cfg.packageOverridden`.

```suggestion
      default = cfg.package.projectName != "apptainer";
      defaultText = literalExpression ''config.services.singularity.package.projectName != "apptainer"'';
```